### PR TITLE
fix: programmatically submit form

### LINKED ISSUE #2104: React 17 submit event not bubbling
Because of the changes in way [react 17 attaches event listeners](https://reactjs.org/blog/2020/10/20/react-v17.html#changes-to-event-delegation), submitting the form programatically no longer works because submit event doesn't bubble.
Solution for this would be to add `bubbles: true` when dispatching CustomEvent submit.

### PR DESCRIPTION
**Pull request by jacqueswho**
### Reasons for making this change

- programmatically submit the form fix

 fixes #2104, #3023

If your PR is non-trivial and you'd like to schedule a synchronous review, please add it to the weekly meeting agenda: https://docs.google.com/document/d/12PjTvv21k6LIky6bNQVnsplMLLnmEuypTLQF8a-8Wss/edit

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X] **I'm adding or updating code**
  - [X] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [X] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/playground/src/app.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import DemoFrame from "./DemoFrame";
 
 const log = (type) => console.log.bind(console, type);
 const toJson = (val) => JSON.stringify(val, null, 2);
+let playGroundForm;
```
Comment (heath-freenome) [inline packages/playground/src/app.js:11]: Can you use the `React.createRef()` function instead?

File: packages/playground/src/app.js [hunk 2]
```diff
@@ -572,6 +581,7 @@ class Playground extends Component {
                 }
                 transformErrors={transformErrors}
                 onError={log("errors")}
+                ref={(form) => {playGroundForm = form;}}
```
Comment (heath-freenome) [inline packages/playground/src/app.js:584]: Pass the `React.createRef()` instance here

File: packages/playground/src/app.js [hunk 3]
```diff
@@ -482,6 +483,14 @@ class Playground extends Component {
               <CopyLink shareURL={this.state.shareURL} onShare={this.onShare} />
             </div>
           </div>
+          <div className="row">
+            <div className="col-sm-10" />
+            <div className="col-sm-2">
+                <button title="Click me to submit the form programmatically." className="btn btn-default" type="button" onClick={()=> playGroundForm.submit()}>
```
Comment (heath-freenome) [inline packages/playground/src/app.js:489]: change `onClick()` to `() => playgroundForm.current.submit()`

File: packages/playground/package.json [hunk 4]
```diff
@@ -110,6 +110,7 @@
     "rimraf": "^3.0.2",
     "sass": "^1.54.5",
     "sass-loader": "^13.0.2",
+    "source-map-loader": "^4.0.0",
```
Comment (heath-freenome) [inline packages/playground/package.json:113]: If you rebase to `master` this and the `package-lock.json` changes will probably go away

**Comment by heath-freenome** [comment]
Also, it looks like you don't have husky setup locally, might I suggest doing `npm run prepare` at the root directory

**Comment by heath-freenome** [comment]
Also, please update the `CHANGELOG.md`

File: packages/playground/src/app.js [hunk 5]
```diff
@@ -482,6 +483,14 @@ class Playground extends Component {
...
+                  Programmatically  Submit
```
Comment (heath-freenome) [inline packages/playground/src/app.js:490]: Also, for a shorter button name, consider `Programatic Submit`

**Comment by heath-freenome** [comment]
@jacqueswho Do you know why the `package-lock.json` changed for core?

**Review by heath-freenome — APPROVED** [review]

File: packages/core/src/components/Form.tsx [hunk 6]
```diff
@@ -713,6 +713,7 @@ export default class Form<T = any, F = any> extends Component<
           cancelable: true,
         })
       );
+      this.formElement.current.requestSubmit();
```
[the remark this answers is not in the document]
Comment (jacqueswho) [reply packages/core/src/components/Form.tsx:716]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/requestSubmit

I see its available on prelease versions of IOS

**Pull request merged**